src/com/google/common/css/MultipleMappingSubstitutionMap.java

```diff
@@ -76,5 +77,5 @@ public static ValueWithMappings createForSingleMapping(String key, String value)
    * associated with this substitution. These mappings are available as
    * {@link ValueWithMappings#mappings}.
    */
-  public ValueWithMappings getValueWithMappings(String key);
+  public ValueWithMappings getValueWithMappings(String key, @Nullable Map<String, String> inputMappings);
```

Comment:
I think I'm not understanding something. If you have the input map like:
```javascript
{"foo": "a", "bar": "b", ...}
```
how can you have an output map that maps the keys differently? My understanding is that we would always return keys from the input mapping when they exist.

I guess I'm thinking of a layering like this:
```
SplittingSubstitutionMap -> RecordingSubstitutionMap -> InputMapSubstitutionMap -> ...
```
This means that an input like `foo-quux` would be split first into `foo` and `quux`, then each would go to the `RecordingSubstitutionMap` which would then pass it to the `InputMapSubstitutionMap`. For `foo`, the `InputSubstitutionMap` would return `a` since it is in the input map. The `RecordingSubstitutionMap` would see this and record `"foo": "a"` in its map. For `quux`, `InputMapSubstitutionMap` would not have an entry in the map, so it would call the next `SubstitutionMap`. Assuming this returns a random new ID, `G8ea1` the `RecordingSubstitutionMap` would see that and record `"quux": "G8ea1"` in its map.

I know that `InputMapSubstitutionMap` doesn't exist right now, but that's what I'm proposing to add.

Does this make sense? What am I missing?

Thanks!